### PR TITLE
javascript: capture arrow functions…

### PR DESCRIPTION
**Pull request by jafl**
…that are not assigned to global variables

### CODE REVIEW

File: parsers/jscript.c [hunk 1]
```diff
@@ -2849,11 +2858,15 @@ static bool parseStatement (tokenInfo *const token, bool is_inside_class)
 			}
 			if (isType (token, TOKEN_CLOSE_CURLY))
 				is_terminated = false;
-
```
Comment (masatake) [inline parsers/jscript.c:2852]: Minor comment. For the original goal of the pull request, you don't have to delete this line. If you want to delete this line, you can make a separate commit just for deleting the line. I have applied this rule to my changes and contributors' changes.
`-p` option of `git add` helps us split a change into two.

However, about the Javascript parser, you are the maintainer. 
So you don't have to follow the rule.
So this comment is just a comment.

**Review by masatake — APPROVED** [review]

File: parsers/jscript.c (hunk 1 above)
Comment (jafl) [reply parsers/jscript.c:2852]: It was a mistake.  I put it back.

**Comment by masatake** [comment]
Let me review this pull request again. I got a question about the life cycle of `sig`.

**Comment by masatake** [comment]
I solved my question.

**Review by masatake — APPROVED** [review]

**Pull request merged**